src/Libraries/Microsoft.Extensions.Diagnostics.ResourceMonitoring/ResourceUtilizationCounters.cs

```diff
@@ -7,24 +7,24 @@ namespace Microsoft.Extensions.Diagnostics.ResourceMonitoring;
 /// Represents the names of instruments published by this package.
 /// </summary>
 /// <remarks>
-/// These counters are currently only published on Linux.
+/// These metrics are currently only published on Linux.
 /// </remarks>
 /// <seealso cref="System.Diagnostics.Metrics.Instrument"/>
 public static class ResourceUtilizationCounters
 {
     /// <summary>
-    /// Gets the CPU consumption of the running application in percentages.
+    /// Gets the CPU consumption of the running application in range <c>[0, 1]</c>.
```

Comment:
I have a stepping back question regarding all utilizations measured by %.

Related discussions and PRs:

* https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/625#discussion_r967319029
* https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/612#discussion_r956469274

FYI @noahfalk IIRC for garbage collector metrics we decided to use absolute value (e.g. total time spent in GC since the beginning of the process lifecycle), and encourage users to derive utilization based on their actual need (so the user can choose whatever sliding window and sampling frequency).

Reply:
As I recall this component has a sampling mechanism and sliding window built into it and the app developer can configure some of those parameters or accept defaults. So when this component reports an instantaneous utilization measurement of 73%, it really means that over the last X minutes some aggregation of Y CPU usage samples had an average CPU utilization of 73% for the app developer's chosen X and Y. Its not the pattern I'd expect most metrics to follow as there is extra implementation complexity and it can't accurately be re-aggregated to compute other sliding windows later. However because the app developer configures a sampling window of known fixed size it doesn't suffer from the issues that those other metrics had where they inferred a sample window from the polling frequency or from some unpredictable process behavior.

A while back I encouraged the engineers who were working on this to simplify and not maintain their own sliding window sampling reservoir, but I feel like what is there now is still well defined and likely useful as long as the sampling window is configured well up-front. I didn't object to this even though I expect metrics of this style will be an outlier in the long run. Having something like this shouldn't block us from later creating an alternate metric that emits absolute measures of CPU and memory usage with more aggregation flexibility.

@reyang do you feel like that addresses your concerns or you are still worried here?

Reply:
I'm not too concerned about having alternative metrics that customers can conveniently use - as long as there is clarity:
1) What exactly does this metric mean.
2) What's the recommendation if there are multiple ways to get similar things - e.g. pros/cons, pitfalls, best practices.

Reply:
> 2. What's the recommendation if there are multiple ways to get similar things - e.g. pros/cons, pitfalls, best practices. For example, https://github.com/open-telemetry/opentelemetry-dotnet-contrib/tree/main/src/OpenTelemetry.Instrumentation.Process.

@noahfalk given OpenTelemetry.Instrumentation.Process is still in Preview, I think we should make a decision about the direction - e.g. all CPU/Memory metrics will be exposed by `Microsoft.Extensions.Diagnostics.ResourceMonitoring`, and we will deprecate `OpenTelemetry.Instrumentation.Process` and point customers to `Microsoft.Extensions.Diagnostics.ResourceMonitoring`. Meanwhile we'll make sure `Microsoft.Extensions.Diagnostics.ResourceMonitoring` works for all supported version of the runtime + operating systems. Does this make sense and do you feel we can make a call before .NET 8 GA?

@Yun-Ting FYI since you've worked on `OpenTelemetry.Instrumentation.Process`.

Reply:
My thought would be:
In .NET 8 - Point devs to these ResourceMonitoring metrics. Either deprecate OTel.Instrumentation.Process or adjust guidance so that we only suggest it only as a fallback for devs that need the metric in a specific form not already handled by ResourceMonitoring.

In the future (.NET 9?) - Implement the core CPU + memory metrics currently in OTel.Instrumentation.Process inside System.Diagnostic.DiagnosticSource. I think metrics like `process.cpu.time` or `process.memory.virtual` are so commonly desired that we shouldn't be asking devs to add extra library dependencies or new background polling services (`serviceCollection.AddResourceMonitoring()`) in order to enable them. At this point ResourceMonitoring would serve two roles going forward: (1) support for networking metrics which are a little more involved to configure (2) back-compat for projects already using the other process metrics defined here and wanting to continue doing so.

What does everyone think? 

Reply:
> My thought would be: In .NET 8 - Point devs to these ResourceMonitoring metrics.

Just confirming - point devs that are using any supported version of .NET (core/framework), not just limited to folks who use .NET 8?

Reply:
Yeah, I meant as of November 2023 (or whenever this assembly releases as stable), all .NET devs should be able to use ResourceMonitoring to satisfy their need for CPU/Virtual Memory metrics.

NOTE: After I wrote that I saw that this library has a weirdly inconsistent implementation of how it measures memory. I'm presuming that gets resolved. If it didn't then I wouldn't feel comfortable recommending it.